sheep/src/lib.rs

```diff
@@ -38,12 +46,22 @@ pub fn pack<P: Packer>(
     stride: usize,
     options: P::Options,
 ) -> Vec<SpriteSheet> {
+    let mut hashes: HashMap<&[u8], usize, BuildHasherDefault<XxHash64>> = Default::default();
+    let mut aliases: Vec<SmallVec<[usize; 1]>> =
+        (0..input.len()).map(|_| SmallVec::new()).collect();
+
+    for (id, sprite) in input.iter().enumerate() {
+        let alias_id = hashes.entry(sprite.bytes.as_slice()).or_insert(id);
+        aliases[*alias_id].push(id);
```

Comment:
If you use `HashaMap::with_capacity` above, you can do `aliases.entry(alias_id).or_insert(SmallVec::new()).push(id)` here.